src/keystore.cpp

```diff
@@ -408,29 +416,13 @@ CBasicKeyStore::GetUFVKMetadataForAddress(const libzcash::UnifiedAddress& addr)
                 if (rmeta.value().GetUFVKId() != ufvkId.value()) {
                     return std::nullopt;
                 }
-
-                if (rmeta.value().GetDiversifierIndex().has_value()) {
-                    if (j.has_value()) {
-                        if (rmeta.value().GetDiversifierIndex().value() != j.value()) {
-                            jConflict = true;
-                            j = std::nullopt;
-                        }
-                    } else if (!jConflict) {
-                        j = rmeta.value().GetDiversifierIndex().value();
-                    }
```

Comment:
I thought there was something that was depending on the `std::nullopt` case here for detecting conflicts; is that not needed?

Reply:
No, there’s no use of the diversifier index from this call on master. My `ALAA` changes would have relied on the diversifier index, but we’re abandoning that.